apps/passport-server/src/database/queries/telegram/fetchTelegramEvent.ts

```diff
@@ -57,21 +57,27 @@ export interface LinkedPretixTelegramEvent {
   telegramChatID: string | null;
   eventName: string;
   configEventID: string;
+  isLinkedToChat: boolean;
 }
 
 export async function fetchLinkedPretixAndTelegramEvents(
-  client: Pool
+  client: Pool,
+  telegramChatId?: number
 ): Promise<LinkedPretixTelegramEvent[]> {
   const result = await sqlQuery(
     client,
-    `\
+    `
     SELECT
       tbe.telegram_chat_id AS "telegramChatID",
       dpe.event_name AS "eventName",
-      dpe.pretix_events_config_id AS "configEventID" 
-    FROM devconnect_pretix_events_info dpe 
-    LEFT JOIN telegram_bot_events tbe ON dpe.pretix_events_config_id = tbe.ticket_event_id
-    `
+      dpe.pretix_events_config_id AS "configEventID",
+      CASE WHEN tbe.telegram_chat_id = $1 THEN true ELSE false END AS "isLinkedToChat"
```

Comment:
Includes whether a chat is linked or not directly in the query instead of mapped on later

Reply:
This is a little confusing with the optional `telegramChatId` param, because it means that when `telegramChatId` is not passed in, `isLinkedToChat` is always false. Also, passing in a `telegramChatId` makes it seem like we are filtering rows by a specific chat id, when we're really not.

Reply:
One thing that might help is renaming the function to `fetchAllTelegramGroups`, the `telegramChatId` param to `currentTelegramChatId`, and then the `isLinkedToChat` field to `isLinkedToCurrentChat`.